src/init.c

```diff
@@ -513,6 +513,13 @@ static void jl_uv_exitcleanup_walk(uv_handle_t *handle, void *arg)
 void jl_write_coverage_data(void);
 void jl_write_malloc_log(void);
 
+static struct uv_shutdown_queue_item *next_shutdown_queue_item(struct uv_shutdown_queue_item *item)
+{
+    struct uv_shutdown_queue_item *rv = item->next;
+    free(item);
+    return rv;
+}
```

Comment:
You mean that valgrind is wrong in the sense that the program is about to exit, so any unfreed memory is about to be cleaned up anyway?  (As far as I can tell there is nothing actually wrong with what valgrind is saying.)  I agree that freeing memory in this case can be considered unnecessary.  But when trying to debug actual memory leaks, having a happy valgrind (made so with such patches, or alternatively with suppressions) makes it a lot easier to spot actual bugs, and I expect will make it much more frequent that people actually run with the leak checker enabled. tl;dr Spurious warnings are no fun.